File: mode.js
// Given a list of numbers, find the mode of the set. 
// Ex: given: [7,3,5,8,3,6,1,3,4,5], mode: 3
// The number 3 occurs three times in this set of integers. This is more than any other integer, so the mode is 3. 
// - Asked about edge cases after

const finMode =(nums)=>{
    let dictNums = {};

    for(const num of nums){
        if(num in dictNums){
            dictNums[num] += 1;
        }else{
            dictNums[num] = 1;
        }
    }

    let maxValue = 0;
    let mode = 0;

    for(const [num, count] of Object.entries(dictNums)){
        if(count > maxValue){
            maxValue = count;
            mode = parseInt(num);
        }
    }

    return mode;
}

console.log(finMode([7,3,5,8,3,6,1,3,4,5]))

File: mode.py
'''
Given a list of numbers, find the mode of the set. 
Ex: given: [7,3,5,8,3,6,1,3,4,5], mode: 3
The number 3 occurs three times in this set of integers. This is more than any other integer, so the mode is 3. 
- Asked about edge cases after
'''

def find_mode(nums):
    is_nums_int = all(isinstance(num, int) for num in nums)
    
    if not is_nums_int or len(nums) <=1:
        return 'all numbers must be integers or the length of numbers must be greater than 1'
    
    dict_nums={}
    
    for num in nums:
        if num in dict_nums:
            dict_nums[num] += 1
        else:
            dict_nums[num] = 1
    maxvalue= 0
    mode = 0
   
    for num, count in dict_nums.items():
        if count > maxvalue:
            maxvalue = count
            mode = num
            
    return mode

print(find_mode([7,3,5,8,3,6,1,3,4,5]))
    
    